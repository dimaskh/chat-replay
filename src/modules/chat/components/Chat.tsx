// Libs
import React from 'react'

// Components
import Header from 'components/Header'
import { MessageList } from 'react-chat-elements'

// Styles
import styled from 'styled-components'

export class Chat extends React.PureComponent<any> {
  public render() {
    return (
      <StyledChat>
        <Header medium>Chat Replay</Header>
        {this.renderContent()}
      </StyledChat>
    )
  }

  private renderContent = (): JSX.Element => {
    const { messages } = this.props

    return <MessageList className="message-list" dataSource={messages} />
  }
}

export const StyledChat = styled.div`
  display: flex;
  flex-direction: column;
  width: 55vw;
  height: 100%;
  color: white;
`

export default Chat
