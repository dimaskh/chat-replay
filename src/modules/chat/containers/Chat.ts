// Redux
import { connect } from 'react-redux'

// Component
import Chat from '../components/Chat'

// Selectors
import { formattedMessages$ } from '../selectors'

// Types
import { StoreState } from 'store/reducer'

const mapStateToProps = (state: StoreState) => ({
  messages: formattedMessages$(state),
})

export default connect(mapStateToProps)(Chat as React.ComponentType<any>)
