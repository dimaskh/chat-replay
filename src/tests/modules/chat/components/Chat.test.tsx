// Libs
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

// Tested component
import Chat from 'modules/chat/components/Chat'

describe('components/Chat', () => {
  const mockMessages = [
    { id: 1, text: 'Message 1' },
    { id: 2, text: 'Message 2' },
  ]

  it('should render correctly', () => {
    const component = shallow(<Chat messages={mockMessages} />)

    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
