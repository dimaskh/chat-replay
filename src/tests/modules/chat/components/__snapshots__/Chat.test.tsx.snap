// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Chat should render correctly 1`] = `
<styled.div>
  <Header
    medium={true}
  >
    Chat Replay
  </Header>
  <t
    className="message-list"
    dataSource={
      Array [
        Object {
          "id": 1,
          "text": "Message 1",
        },
        Object {
          "id": 2,
          "text": "Message 2",
        },
      ]
    }
    downButton={true}
    downButtonBadge={null}
    lockable={false}
    onClick={null}
    onDownButtonClick={null}
    onDownload={null}
    onForwardClick={null}
    onOpen={null}
    onPhotoError={null}
    onTitleClick={null}
    toBottomHeight={300}
  />
</styled.div>
`;
